server/api/crud/artifacts.py

```diff
@@ -227,3 +250,49 @@ def _resolve_artifact_size(artifact, auth_info):
                     )
         if "spec" in artifact and "inline" in artifact["spec"]:
             validate_inline_artifact_body_size(artifact["spec"]["inline"])
+
+    def _delete_artifact_data(
+        self,
+        db_session: sqlalchemy.orm.Session,
+        key: str,
+        tag: str = "latest",
+        project: str = mlrun.mlconf.default_project,
+        object_uid: str = None,
+        producer_id: str = None,
+        deletion_strategy: mlrun.common.schemas.artifact.ArtifactsDeletionStrategies = (
+            mlrun.common.schemas.artifact.ArtifactsDeletionStrategies.metadata_only
+        ),
+        secrets: dict = None,
+        auth_info: mlrun.common.schemas.AuthInfo = mlrun.common.schemas.AuthInfo(),
+    ):
+        logger.debug("Deleting artifact data", project=project, key=key, tag=tag)
+
+        try:
+            artifact = self.get_artifact(
+                db_session,
+                key,
+                tag,
+                project=project,
+                producer_id=producer_id,
+                object_uid=object_uid,
+            )
+            path = artifact["spec"]["target_path"]
```

Comment:
is this needed or will you get the artifact from the client?

Reply:
I need the artifact target path - the DELETE request get the artifact key as parameter and not the artifact itself